mgmt/MgmtDefs.h

```diff
@@ -72,71 +73,78 @@ struct MgmtConverter {
    * This is passed a @c void* which is a pointer to the member in the configuration instance.
    * This function must return a @c MgmtInt converted from that value.
    */
-  MgmtInt (*load_int)(const void *) = nullptr;
+  MgmtInt (*load_int)(const TSHttpTxn txnp, const void *) = nullptr;
```

Comment:
In order to use, you would need to reinterpret cast it to HttpSM * no?  It seems like it would be better to have the parameter type be HttpSM *.  I think rule of thumb is that apidefs.h and ts.h should only be included in plugins and the implementation files for the TS API.  Unfortunately, I think you will probably not be able to use const HttpSM *, I don't think the HttpSM class is const correct.

Reply:
Walt is basically correct. Originally there was just "ts.h". I split it into "ts.h" and "apidefs.h" precisely so "apidefs.h" could be included in core code. Core code should never include "ts.h". Core code should use C Plugin API types only in API functions expected to be called from a plugin.